runtime/parachains/src/inclusion_inherent.rs

```diff
@@ -127,6 +129,30 @@ decl_module! {
 	}
 }
 
+// At the point that this is run, the provisioner has already chosen a set of extrinsics,
+// so we can rely on calculations like block weight.
```

Comment:
I _think_ this assertion is true, but would appreciate someone more knowledgable confirming or denying it.

Reply:
I don't think this is necessarily true.

Reply:
as far as I know, for now all inherents are run before non-inherent extrinsics. thus the block weight here only includes on_runtime_upgrade, on_initialize weights and previous inherent weights.